cmd/app/createca.go

```diff
@@ -19,22 +19,40 @@ import (
 	"crypto"
 	"crypto/ecdsa"
 	"crypto/rand"
+	"crypto/rsa"
 	"crypto/x509"
 	"crypto/x509/pkix"
 	"encoding/pem"
+	"errors"
 	"fmt"
 	"math"
 	"math/big"
 	"os"
 	"time"
 
 	"github.com/ThalesIgnite/crypto11"
+	"github.com/sigstore/fulcio/pkg/log"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+)
 
-	"github.com/sigstore/fulcio/pkg/log"
+const (
+	label = "FulcioCA"
 )
 
+func publicKeyFromPrivate(privKey crypto.PrivateKey) (crypto.PublicKey, error) {
+	var err error
+	switch key := privKey.(type) {
```

Comment:
you're returning a `crypto.PublicKey` here, which means there's no point in casting the return value to the algorithm-specific value. This would work just the same:

```
switch key := privKey.(type) {
case *ecdsa.PrivateKey, *rsa.PrivateKey:
	return key.Public(), nil
default:
	return nil, errors.New("error generating public key")
}
```

Reply:
This does not work, having the case match both leave Key undefined:

`cmd/app/createca.go:49:13: key.Public undefined (type crypto.PrivateKey is interface with no methods)`

Reply:
I get what you mean here, in fact I don't need to any checks at all.

this works fine: `pubKey := privKey.Public().(crypto.PublicKey)`

This also made me check rsa i the HSM and its all good.